### PR TITLE
MGDOBR-816: Add Data Plane dependencies installation scripts

### PR DESCRIPTION
**Pull request by winklerm**
[MGDOBR-816](https://issues.redhat.com/browse/MGDOBR-816)

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: scripts/dependencies/servicemesh/kialiSub.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+kind: Subscription
+metadata:
+  name: openshift-servicemesh-kiali-operator-subscription
+  namespace: openshift-operators
+spec:
+  channel: stable
+  name: kiali-ossm
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  installPlanApproval: Manual
```
Comment (msmagnanijr) [inline scripts/dependencies/servicemesh/kialiSub.yaml:11]: Please add:  startingCSV: kiali-operator.v1.48.2

File: scripts/dependencies/servicemesh/serviceMeshSub.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+kind: Subscription
+metadata:
+  name: openshift-servicemesh-operator-subscription
+  namespace: openshift-operators
+spec:
+  channel: stable
+  name: servicemeshoperator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  installPlanApproval: Manual
```
Comment (msmagnanijr) [inline scripts/dependencies/servicemesh/serviceMeshSub.yaml:11]: Please add:   startingCSV: servicemeshoperator.v2.2.2

File: scripts/dependencies/servicemesh/tracingSub.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+kind: Subscription
+metadata:
+  name: openshift-servicemesh-tracing-operator-subscription
+  namespace: openshift-operators
+spec:
+  channel: stable
+  name: jaeger-product
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  installPlanApproval: Manual
```
Comment (msmagnanijr) [inline scripts/dependencies/servicemesh/tracingSub.yaml:11]: Please add:   startingCSV: jaeger-operator.v1.36.0-2

**Review by msmagnanijr — CHANGES_REQUESTED** [review]

File: scripts/dependencies/servicemesh/kialiSub.yaml (hunk 1 above)
Comment (winklerm) [reply scripts/dependencies/servicemesh/kialiSub.yaml:11]: Sure, good point!

File: scripts/dependencies/servicemesh/serviceMeshSub.yaml (hunk 2 above)
Comment (winklerm) [reply scripts/dependencies/servicemesh/serviceMeshSub.yaml:12]: Done!

File: scripts/dependencies/servicemesh/kialiSub.yaml (hunk 1 above)
Comment (winklerm) [reply scripts/dependencies/servicemesh/kialiSub.yaml:12]: Done!

**Comment by winklerm** [comment]
I have just realized that the Manual `installPlanApproval` needs to be approved (as described in [1]), will add it.

[1] https://olm.operatorframework.io/docs/tasks/install-operator-with-olm/

File: scripts/dependencies/install_dependencies.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+echo "Installing ServiceMesh Operators"
+# according to https://docs.openshift.com/container-platform/4.11/service_mesh/v2x/installing-ossm.html
+# the individual operators have to be installed in this order and each one only after the previous one's installation finished
+install_operator_and_wait servicemesh/tracingSub.yaml
+install_operator_and_wait servicemesh/kialiSub.yaml
+install_operator_and_wait servicemesh/serviceMeshSub.yaml
+
+# set up service mesh resources
+oc apply -k servicemesh
```
[the remark this answers is not in the document]
Comment (winklerm) [reply scripts/dependencies/install_dependencies.sh:19]: Indeed, we have agreed with @vaibhavjainwiz to put those scripts in this PR until his change is ready - but it seems it was already merged 2 days ago. Please let me know if I should remove it, thanks!

[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply scripts/dependencies/install_dependencies.sh:19]: Hi @winklerm 
We could remove istio related script now. It is handled by operator now.

[the remark this answers is not in the document]
Comment (winklerm) [reply scripts/dependencies/install_dependencies.sh:19]: Great, will remove it! One question, are all the resources in servicemesh/resources [1] handled by the operator (incl. the `istio-system` namespace creation)?

[1] https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1134/files#diff-86f35a740dda1558c43e4322904f4762f1b22f474a01d991b1db30ba66afba23

[the remark this answers is not in the document]
Comment (winklerm) [reply scripts/dependencies/install_dependencies.sh:19]: Removed the service mesh resources, except for `istio-system` namespace, servicemesh control plane and service mesh member roll.

**Review by msmagnanijr — APPROVED** [review]
/lgtm

**Comment by winklerm** [comment]
Added the plan approval and uninstallation script, tested against our OSD cluster, console output is here [1].

[1] http://pastebin.test.redhat.com/1073877

**Review by vaibhavjainwiz — APPROVED** [review]
lgtm

**Pull request merged**